src/commons/api/aspa.rs

```diff
@@ -158,26 +163,29 @@ impl FromStr for AspaDefinition {
 
         let mut providers = {
             let mut providers = vec![];
-            let providers_str = parts.next().ok_or(AspaDefinitionFormatError::ProviderAsMissing)?;
-            let provider_parts = providers_str.split(',');
-            for provider_part in provider_parts {
-                let provider = ProviderAs::from_str(provider_part.trim())
-                    .map_err(|_| AspaDefinitionFormatError::ProviderAsInvalid(provider_part.trim().to_string()))?;
-                providers.push(provider);
+            let providers_str = parts.next().unwrap_or("<none>");
+
+            if providers_str.trim() != "<none>" {
+                let provider_parts = providers_str.split(',');
+                for provider_part in provider_parts {
+                    let provider = ProviderAs::from_str(provider_part.trim())
+                        .map_err(|_| AspaDefinitionFormatError::ProviderAsInvalid(provider_part.trim().to_string()))?;
+                    providers.push(provider);
+                }
             }
+
             providers
         };
 
         // unexpected extra bits are not acceptable
         if parts.next().is_some() {
             Err(AspaDefinitionFormatError::ExtraParts)
         } else {
-            // Ensure that the providers are sorted, there is at least one, and there are no duplicates
+            // Ensure that the providers are sorted,  and there are no duplicates
             providers.sort_by_key(|p| p.provider());
 
-            let mut last_seen = providers.first().ok_or(AspaDefinitionFormatError::ProviderAsMissing)?;
-
             if providers.len() > 1 {
+                let mut last_seen = providers.first().unwrap();
```

Comment:
This is probably better done using `providers.windows(2).find(...)` or somesuch?

Reply:
yes, good point. Much improved now I hope.